src/schema/objects/formats.yaml

```diff
@@ -0,0 +1,99 @@
+---
+# This file defines valid patterns for different formats
+# Entity patterns
+label:
+  description: |
+    Freeform labels without special characters.
+  pattern: ^[0-9a-zA-Z]+$
```

Comment:
what is the motivation for explicit anchoring?
We with @TheChymera we are thinking to establish regex based validation of BIDS filenames for DANDI project... for that, we likely just to compose a list of regexes for a valid BIDS filename.  Then, `<label>` would be replaced with regex from here.  If it is anchored -- it would require us to strip those first.  I think anchors should be explicitly added in the code whenever processing already "extracted" label.

Reply:
I believe I kept the anchors from the validator expressions. I can definitely remove them here, and then the validator would just add them back I think.

Reply:
Are quotes a problem? Some of the regular expressions seem to cause problems with YAML if I don't wrap them in quotes.